src/btree/bt_sync.c

```diff
@@ -143,6 +143,42 @@ __sync_ref_mark_deleted(WT_SESSION_IMPL *session, WT_REF *ref)
     return (0);
 }
 
+/*
+ * __sync_ref_int_obsolete_cleanup --
+ *     Traverse the internal page and identify the leaf pages that are obsolete and mark them as
+ *     deleted.
+ */
+static int
+__sync_ref_int_obsolete_cleanup(WT_SESSION_IMPL *session, WT_REF *intref)
+{
+    WT_PAGE_INDEX *pindex;
+    WT_REF *ref;
+    uint32_t slot;
+
+    pindex = NULL;
+
+    pindex = WT_INTL_INDEX_GET_SAFE(intref->page);
```

Comment:
I'm confused why we're using `WT_INTL_INDEX_GET_SAFE` instead of `WT_INTL_INDEX_GET`, what makes this call safe?

Reply:
Thanks, now I understood. I used the `WT_WITH_PAGE_INDEX` to protect it.